packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -116,9 +116,11 @@ export class SerializedStateManager {
 					eventName: "getSnapshotTreeFailed",
 					id: version.id,
 				});
-			} else if (snapshot !== undefined && version === undefined) {
+				// Not sure if this should be here actually
+			} else if (snapshot !== undefined && version?.id === undefined) {
 				this.mc.logger.sendErrorEvent({
-					eventName: "getSnapshotFetchedTreeWithoutVersion",
+					eventName: "getSnapshotFetchedTreeWithoutVersionId",
+					hasVersion: version !== undefined, // if hasVersion is true, this means that the contract with the service was broken.
```

Comment:
I do not understand how version can be undefined, given like 109 where it's set, and it's const.

Reply:
After doing a deep investigation, I strongly believe that this is the case for Tinylicious in a very rare case. I think it's some sort of race condition. Considering we are trying to move to `WholeSummaryUploadManager`, it may disappear then, but this should catch it.

Reply:
Oh wait, let me fix that

Reply:
Indeed tinylicious e2e tests [just hit this](https://dev.azure.com/fluidframework/internal/_build/results?buildId=245132&view=logs&j=60818a5b-4587-599f-d5b2-9fbcf26f5776&t=958c8502-08fb-53f8-797c-adcae3695e13&l=4720).

Reply:
[Once more in e2e tests](https://dev.azure.com/fluidframework/internal/_build/results?buildId=245278&view=logs&j=60818a5b-4587-599f-d5b2-9fbcf26f5776&t=958c8502-08fb-53f8-797c-adcae3695e13&l=4718). Seems like it won't be hard to repro.

Reply:
This makes me happy :)

Reply:
The isomorphic-git fix [turned out not to be that complicated](https://github.com/isomorphic-git/isomorphic-git/pull/1882). If that gets merged and released soon, the path forward for us is to simply update our deps on it.